nodkz/mongodb-memory-server
# Fix/various

- replace "npm" with "yarn" to run scripts
- rename function "getUriBase" to "uriTemplate"
- only add "?" when query options are provided
- add test file "db_util-test"
- remove test "v6.0.0 adds "?" to the connection string (uri)"
- add regex to test the message of the error to ensure that is the only error ignored
- rename all test files from "-test" to ".test"
- change "cache" to be an Map
- change "getCachePath" return type to "string | undefined"
- remove default for variable "binaryPath"
- change interfaces & types to have an unified naming-style (remove "T")
- add section "Code Styles" to CONTRIBUTING
- uriTemplate: change "query" to be an string-array
- remove file "types.ts"
- move all types from "type.ts" into appropriate files
- rename interface "DownloadProgress" to "MongoBinaryDownloadProgress"
- style(MongoInstance): add TODO comments
- rename to match other file names [camelCase]
  - rename file "db_util" to "utils"
  - rename file "db_util.test" to "utils.test"
  - rename variables "db_util" to "utils"
  - rename file "resolve-config" to "resolveConfig"
  - rename file "resolve-config.test" to "resolveConfig.test"
  - rename file "postinstall-helper" to "postinstallHelper"
- chore(CONTRIBUTING): add section "File Names"
- test(MongoBinaryDownloadUrl): fix spies & mocks
- feat(MongoMemoryServer): add ability to automatically create auth
- feat(MongoMemoryReplSet): add basic "createAuth" support
- fix(utils): killProcess: check if the childProcess PID is still alive
- feat(MongoMemoryServer): add (protected) function "getNewPort"
- ensureInstance: move "return instanceInfo" into case "running"
- feat(MongoMemoryServer): add function "getStartOptions"
- refactor(MongoMemoryServer): start: check "state" instead of "instanceInfo"
- move test from "singleDB" to "MongoMemoryServer"
- feat(MongoMemoryServer): allow re-use of instances & dbPath
- fix(MongoInstance): run: reset all booleans on
- feat(MongoMemoryServer): add ability to force same port on restart
- fix(MongoMemoryReplSet): _waitForPrimary: check if instance is already primary
- feat(MongoMemoryReplSet): getUri: allow executing while state is "init"
- initAllServers: execute "stateChange" with "init"
- feat(MongoMemoryReplSet): allow re-use of instances & dbPath

## Related Issues

- closes #404 
- closes #374 
- closes #395 
- closes #406 
- closes #299
